src/extremeweatherbench/regions.py

```diff
@@ -173,29 +217,66 @@ class ShapefileRegion(Region):
     on instantiation.
 
     Attributes:
-        shapefile_path: Path to the shapefile
+        shapefile_path: Local or remote path to the .shp shapefile
     """
 
     def __repr__(self):
         return f"{self.__class__.__name__}(shapefile_path={self.shapefile_path})"
 
-    def __init__(self, shapefile_path: str | Path):
-        self.shapefile_path = Path(shapefile_path)
+    def __init__(self, shapefile_path: str | pathlib.Path):
+        self.shapefile_path = pathlib.Path(shapefile_path)
 
     @classmethod
-    def create_region(cls, shapefile_path: str | Path) -> "ShapefileRegion":
+    def create_region(cls, shapefile_path: str | pathlib.Path) -> "ShapefileRegion":
         """Create a ShapefileRegion with the given parameters."""
         return cls(shapefile_path=str(shapefile_path))
 
     @property
     def geopandas(self) -> gpd.GeoDataFrame:
-        """Return representation of this Region as a GeoDataFrame."""
+        """Return representation of this Region as a GeoDataFrame.
+
+        Returns:
+            A GeoDataFrame representing the region.
+        """
         try:
             return gpd.read_file(self.shapefile_path)
         except Exception as e:
             logger.error(f"Error reading shapefile: {e}")
             raise ValueError(f"Error reading shapefile: {e}")
 
+    def mask(self, dataset: xr.Dataset, drop: bool = False) -> xr.Dataset:
+        """Mask a dataset to the region.
+
+        Args:
+            dataset: The dataset to mask.
+            drop: Whether to drop NaN values outside the region. Defaults to False.
+
+        Returns:
+            The subset dataset.
+        """
+        longitude_min, latitude_min, longitude_max, latitude_max = (
+            self.geopandas.total_bounds
+        )
+        try:
+            dataset = dataset.sel(
+                latitude=slice(latitude_min, latitude_max),
+                longitude=slice(longitude_min, longitude_max),
+                drop=drop,
+            )
+        except Exception:
+            # If the latitude slice fails, try the reverse
+            dataset = dataset.sel(
+                latitude=slice(latitude_max, latitude_min),
+                longitude=slice(longitude_min, longitude_max),
+                drop=drop,
+            )
+        # Subset dataset after cutting out a box to minimize memory pressure
+        mask = regionmask.mask_geopandas(
+            self.geopandas, dataset.longitude, dataset.latitude
+        )
+        mask_array = ~np.isnan(mask)
+        return dataset.where(mask_array, drop=drop)
```

Comment:
Unless you plan on doing something with `mask_array`, stylistically it's a bit more readable to simply apply it in-line with the return statement.

```suggestion
        return dataset.where(~np.isnan(mask), drop=drop)
```